components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatient.java

```diff
@@ -325,7 +333,8 @@ public JSONObject toJSON(Collection<String> onlyFieldNames)
             result.element(JSON_KEY_DISORDERS, diseasesToJSON());
         }
 
-        for (PatientDataController<?> serializer : this.serializers) {
+
+        for (PatientDataController<?> serializer: this.serializers.values()) {
```

Comment:
A `TreeMap` requires that the values are comparable, which isn't the case for `PatientData`.